proposals/004-enable-package-installation-from-branch.md

```diff
@@ -0,0 +1,101 @@
+# Enable package installation from branch
+
+**Closing date:** 2018-11-02
+
+**Status:** proposed
+
+## Summary
+
+Enable consumption of GOV.UK Frontend from a repository branch by refactoring
+the folder structure and updating the automated scripts.
+
+Example of npm installation instructions from a branch:
+
+`npm install --save alphagov/govuk-frontend#branch-name`
+
+## Problem
+
+Currently it is not possible to consume an unreleased version of GOV.UK Frontend.
+
+There are three main reasons for wanting to do that:
+1. You want to integrate a component that is work in progress into your project.
+This is useful for getting early feedback on a component through user research
+especially when a project is in early stages (i.e. alpha or private beta)
+
+1. You want to prepare guidance and examples in the Design System for a component
+that is in the process of contribution.
+
+1. You want to test an unreleased version of GOV.UK Frontend in the Design System.
```

Comment:
I believe this last one is not a big barrier right now using `npm link`

```bash
cd ../govuk-design-system
git checkout master
git pull
npm install # note running `npm install` after `npm link` will destroy the link.
npm link ../govuk-frontend/package/
```

I don't believe that npm works for the other cases you have mentioned though.

https://github.com/alphagov/govuk-frontend/blob/master/docs/contributing/publishing.md

Reply:
Yes, it can be avoided by using `npm link` or even a symlink to a local dir, but a content designer or a designer might find it harder. The rest of the process can be easily done through the github UI.